x-pack/plugins/apm/public/context/UrlParamsContext/index.tsx

```diff
@@ -27,6 +28,16 @@ interface TimeRangeRefreshAction {
   time: TimeRange;
 }
 
+function getUIFilters(urlParams: IUrlParams): UIFilters {
+  return useMemo(
+    () => ({
+      kuery: urlParams.kuery,
+      environment: urlParams.environment
+    }),
+    [urlParams]
+  );
+}
+
```

Comment:
Isn't this a hook, since it uses `useMemo` under the hood? And thus should be called `useUIFilters`?
I'm asking because hooks must be called unconditionally (if you try calling them conditionally React should `throw` but worst case it'll silently lead to really weird states). Therefore it's very helpful if hooks are declared as such, so it's easy to see.

Note: I _think_ `useMemo` is a hook which much be called unconditionally but I'm not certain.

Reply:
You're right, maybe i should use the lodash _.memoize function instead with urlParams as the cache key?

Reply:
That could work - although I think `useMemo` is also fine - it's just about setting the expectation for consumers.